pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go

```diff
@@ -188,12 +194,13 @@ func (nc *NamespaceController) syncNamespace(ns *v1.Namespace) {
 	if nc.DiscoveryNamespacesFilter != nil && !nc.DiscoveryNamespacesFilter.FilterNamespace(ns.ObjectMeta) {
 		return
 	}
+	nc.queue.Add(types.NamespacedName{Name: ns.Name})
+}
 
-	// If a MemberRoll controller is in use, and the set of
-	// namespaces still includes the one for this ConfigMap,
-	// then recreate the ConfigMap, otherwise do nothing.
-	if nc.usesMemberRollController && !nc.namespaces.Contains(ns.Name) {
+func (nc *NamespaceController) syncNamespaceByName(ns string) {
```

Comment:
by creating a new `func` here we're skipping the DiscoveryNamespacesFilter that is present in `syncNamespace()`. That would be a behavior change, so I'd suggest we either merge the two functions again or add the filtering logic here as well

Reply:
I did it intentionally, because we cannot implement analogous function based on NamespeceSet from xns-informer, which is a list of namespace names, not namespace objects, so we cannot perform filtering of label selectors.
I ignored this filtering, because our users select namespaces with SMMR, not discovery selectors. Of course they could use SMMR + discovery selectors, but it can only be configured with `meshConfig.discoverySelectors`, which is tech-preview or an unsupported API. So should we care about this filtering?

Reply:
I think that I didn't break anything, because namespace lister is not initialized when SMMR controller is enabled, so that function couldn't work in multi-tenant mode anyway. I just found an execution path that fails.

Reply:
oh you're right, we dont have access to the labels anyway. ignore my comment